specs/PeriodicBackgoundSyncAndBackgroundSync.md

```diff
@@ -0,0 +1,581 @@
+                            CHECK_FAILURE(m_syncRegistrationManager->add_BackgroundSyncRegistered(
+                                Microsoft::WRL::Callback<
+                                    ICoreWebView2SyncRegisteredEventHandler>(
+                                    [this](
+                                        ICoreWebView2ServiceWorkerSyncRegistrationManager*
+                                            sender,
+                                        ICoreWebView2ServiceWorkerSyncRegisteredEventArgs*
+                                            args)
+                                    {
+                                        wil::com_ptr<
+                                            ICoreWebView2ServiceWorkerSyncRegistrationInfo>
+                                            syncRegistrationInfo;
+                                        CHECK_FAILURE(
+                                            args->get_RegistrationInfo(&syncRegistrationInfo));
+                                        std::wstringstream message;
+                                        AppendSyncRegistrationInfo(
+                                            syncRegistrationInfo, false, message);
+                                        m_appWindow->AsyncMessageBox(
+                                            message.str(), L"Background Sync");
+
+                                        return S_OK;
+                                    })
+                                    .Get(),
+                                &m_backgroundSyncRegisteredToken));
+                            //! [BackgroundSyncRegistered]
+                            //! [PeriodicSyncRegistered]
+                            CHECK_FAILURE(m_syncRegistrationManager->add_PeriodicSyncRegistered(
+                                Microsoft::WRL::Callback<
+                                    ICoreWebView2SyncRegisteredEventHandler>(
+                                    [this](
+                                        ICoreWebView2ServiceWorkerSyncRegistrationManager*
+                                            sender,
+                                        ICoreWebView2ServiceWorkerSyncRegisteredEventArgs*
+                                            args)
+                                    {
+                                        wil::com_ptr<
+                                            ICoreWebView2ServiceWorkerSyncRegistrationInfo>
+                                            syncRegistrationInfo;
+                                        CHECK_FAILURE(
+                                            args->get_RegistrationInfo(&syncRegistrationInfo));
+                                        std::wstringstream message;
+                                        AppendSyncRegistrationInfo(
+                                            syncRegistrationInfo, true, message);
+                                        m_appWindow->AsyncMessageBox(

```

Comment:
We want the sample code to demonstrate real scenarios for the API. In this case it looks like you are just showing a message box when the event is raised. Is there a more realistic scenario where the end dev would be using this event for something that you can show instead? (If this leads to a lot of code that is unrelated to the API then you can call into a helper method that you don't define in the sample code with a comment explaining what the helper would do)